tests/test_connector_sharded.py

```diff
@@ -29,34 +30,63 @@
 class TestConnectorSharded(unittest.TestCase):
 
     def setUp(self):
-        if not (db_user and db_password):
-            raise SkipTest('Need to set a user/password to test this.')
+        if db_user and db_password:
+            auth_args = dict(auth_username=db_user, auth_key=db_password)
+        else:
+            auth_args = {}
         self.cluster = ShardedClusterSingle().start()
+        self.dm = DocManager()
+        self.connector = Connector(
+            mongo_address=self.cluster.uri,
+            doc_managers=[self.dm],
+            **auth_args
+        )
+        self.connector.start()
```

Comment:
Refactored `TestConnectorSharded` to `setUp` the `Connector` used by both `TestConnectorShardedMongosFailure` and `TestConnectorShardedAuth`.